Code/Source/Lidar/ROS2LidarSensorComponent.cpp

```diff
@@ -102,14 +104,21 @@ namespace ROS2
         m_pointCloudPublisher.reset();
     }
 
-    void ROS2LidarSensorComponent::FrequencyTick()
-    {
+    void ROS2LidarSensorComponent::FrequencyTick() {
         float distance = LidarTemplateUtils::GetTemplate(m_lidarModel).m_maxRange;
         auto entityTransform = GetEntity()->FindComponent<AzFramework::TransformComponent>(); // TODO - go through ROS2Frame
-        const auto directions = LidarTemplateUtils::PopulateRayDirections(m_lidarModel, entityTransform->GetWorldTM().GetEulerRadians());
+        const auto directions = LidarTemplateUtils::PopulateRayDirections(m_lidarModel,
+                                                                          entityTransform->GetWorldTM().GetEulerRadians());
         AZ::Vector3 start = entityTransform->GetWorldTM().GetTranslation();
-        start.SetZ(start.GetZ() + 1.0f);
-        m_lastScanResults = m_lidarRaycaster.PerformRaycast(start, directions, distance);
+        start.SetZ(start.GetZ());
+
+        // If self collision entity is not set, use this one
+        if (!m_selfColliderEntityId.IsValid())
```

Comment:
would be best to communicate to the user that this is optional - or set the self by default. I will design the feature.

Reply:
I added a hint to the description